### PR TITLE
fixes issue/57

### PR DESCRIPTION
**Pull request by NayanKhedkar**
replaced node-native-zip-compression with archiver

### CODE REVIEW

File: commands/045zip.js [hunk 1]
```diff
@@ -72,21 +74,30 @@ commands.create({
...
 
-      archive.addFiles(zipFiles, function (err) {
-        if (err) {
-          warn(namePrefix + err)
-          resolve()
-          return
-        }
+      output.on('close', () => {
+        resolve()
+      });
```
Comment (oliverfoster) [inline commands/045zip.js:82]:
```suggestion
      output.on('close', resolve)
```

File: commands/045zip.js [hunk 2]
```diff
@@ -72,21 +74,30 @@ commands.create({
...
-        archive.toBuffer(function (buff) {
-          const fileName = path.join(outputDir, scoDate + '_' + name.replace(/[|&;$%@"<>()/\\+,]/g, '_') + '.zip')
-          fs.writeFile(fileName, buff, function () {
-            resolve()
-          })
-        })
-      })
+      output.on('end', () => {
+        resolve()
+      });
```
Comment (oliverfoster) [inline commands/045zip.js:86]:
```suggestion
      output.on('end', resolve)
```

File: commands/045zip.js [hunk 3]
```diff
@@ -72,21 +74,30 @@ commands.create({
...
+      archive.pipe(output)
+
+      zipFiles.forEach((file) => {
+        archive.append(fs.createReadStream(file.path), { name: file.name })
+      });
+
+      archive.on('error', (err) => {
+        log(err)
+        resolve()
+      });
```
Comment (oliverfoster) [inline commands/045zip.js:98]:
```suggestion
      zipFiles.forEach(file => {
        archive.append(fs.createReadStream(file.path), { name: file.name })
      })

      archive.on('error', err => {
        log(err)
        resolve()
      })
```

File: commands/045zip.js [hunk 4]
```diff
@@ -59,7 +59,9 @@ commands.create({
       dirs: false
     })
     return new Promise((resolve, reject) => {
-      const archive = new ZipLibrary()
+      const archive = archiver('zip', {
+        zlib: { level: 9 } // Sets the compression level.
+      });
```
Comment (oliverfoster) [inline commands/045zip.js:64]:
```suggestion
      })
```

File: commands/046zipaat.js [hunk 5]
```diff
@@ -68,7 +68,9 @@ commands.create({
       dirs: false
     })
     return new Promise((resolve, reject) => {
-      const archive = new ZipLibrary()
+      const archive = archiver('zip', {
+        zlib: { level: 9 } // Sets the compression level.
+      });
```
Comment (oliverfoster) [inline commands/046zipaat.js:73]:
```suggestion
      })
```

File: commands/046zipaat.js [hunk 6]
```diff
@@ -84,21 +86,28 @@ commands.create({
         resolve()
         return
       }
+      const output = fs.createWriteStream(path.join(outputDir, scoDate + '_' + name.replace(/[|&;$%@"<>()/\\+,]/g, '_') + '.aat.zip'))
+      
+      output.on('close', () => {
+        resolve()
+      });
```
Comment (oliverfoster) [inline commands/046zipaat.js:93]:
```suggestion
      output.on('close', resolve)
```

File: commands/046zipaat.js [hunk 7]
```diff
@@ -84,21 +86,28 @@ commands.create({
...
 
-      archive.addFiles(zipFiles, function (err) {
-        if (err) {
-          warn(namePrefix + err)
-          resolve()
-          return
-        }
+      output.on('end', () => {
+        resolve()
+      });
```
Comment (oliverfoster) [inline commands/046zipaat.js:97]:
```suggestion
      output.on('end', resolve)
```

File: commands/046zipaat.js [hunk 8]
```diff
@@ -84,21 +86,28 @@ commands.create({
...
-          fs.writeFile(fileName, buff, function () {
-            resolve()
-          })
-        })
-      })
+      archive.pipe(output)
+
+      zipFiles.forEach((file)=>{
+        archive.append(fs.createReadStream(file.path), { name: file.name })
+      });
```
Comment (oliverfoster) [inline commands/046zipaat.js:103]:
```suggestion
      zipFiles.forEach(file => {
        archive.append(fs.createReadStream(file.path), { name: file.name })
      })
```

File: commands/046zipaat.js [hunk 9]
```diff
@@ -84,21 +86,28 @@ commands.create({
...
+
+      archive.on('error', (err) => {
+        log(err)
+        resolve()
+      });
```
Comment (oliverfoster) [inline commands/046zipaat.js:108]:
```suggestion
      archive.on('error', err => {
        log(err)
        resolve()
      })
```

**Review by oliverfoster — CHANGES_REQUESTED** [review]
Remove line end `;` semi colons
Remove unnecessary brackets from single argument fat-arrow  functions
Remote unnecessary fat arrow functions

**Comment by NayanKhedkar** [comment]
implemented requested changes

**Review by oliverfoster — APPROVED** [review]

**Pull request merged**